test/e2e/installplan_e2e_test.go

```diff
@@ -2998,6 +2999,143 @@ var _ = Describe("Install Plan", func() {
+					Image:      "quay.io/olmtest/single-bundle-index:1.0.0",
+					SourceType: operatorsv1alpha1.SourceTypeGrpc,
+				},
+			}
+			Expect(ctx.Ctx().Client().Create(context.Background(), catsrc)).To(Succeed())
+
+			// Wait for the CatalogSource to be ready
+			catsrc, err := fetchCatalogSourceOnStatus(crc, catsrc.GetName(), catsrc.GetNamespace(), catalogSourceRegistryPodSynced)
+			Expect(err).ToNot(HaveOccurred())
+
+			catsrcName = catsrc.GetName()
+
+			// Create the OperatorGroup
+			og := &operatorsv1.OperatorGroup{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "og",
+					Namespace: ns.GetName(),
+				},
+				Spec: operatorsv1.OperatorGroupSpec{
+					TargetNamespaces: []string{ns.GetName()},
+				},
+			}
+			Expect(ctx.Ctx().Client().Create(context.Background(), og)).To(Succeed())
+
+			now := metav1.Now()
+			ip = &operatorsv1alpha1.InstallPlan{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "ip",
+					Namespace: ns.GetName(),
+				},
+				Spec: operatorsv1alpha1.InstallPlanSpec{
+					ClusterServiceVersionNames: []string{"foobar"},
+					Approval:                   v1alpha1.ApprovalAutomatic,
+					Approved:                   true,
+				},
+				Status: operatorsv1alpha1.InstallPlanStatus{
+					Phase:          operatorsv1alpha1.InstallPlanPhasePlanning,
+					CatalogSources: []string{},
+					BundleLookups: []operatorsv1alpha1.BundleLookup{
+						{
+							Identifier: "foobar.v0.0.1",
+							CatalogSourceRef: &corev1.ObjectReference{
+								Namespace: ns.GetName(),
+								Name:      catsrcName,
+							},
+							Conditions: []operatorsv1alpha1.BundleLookupCondition{
+								{
+									Type:               operatorsv1alpha1.BundleLookupPending,
+									Status:             corev1.ConditionTrue,
+									Reason:             "JobIncomplete",
+									Message:            "unpack job not completed",
+									LastTransitionTime: &now,
+								},
+							},
+						},
+					},
+				},
+			}
+		})
+
+		AfterEach(func() {
+			Expect(ctx.Ctx().Client().Delete(context.Background(), ns)).To(Succeed())
+		})
+
+		It("should timeout and fail the InstallPlan for a non-existent bundle image", func() {
+			// Create an InstallPlan status.bundleLookups.Path specified for a non-existent bundle image
+			ip.Status.BundleLookups[0].Path = "quay.io/foo/bar:v0.0.1"
+
+			outIP := ip.DeepCopy()
+
+			Expect(ctx.Ctx().Client().Create(context.Background(), outIP)).To(Succeed())
+
+			// The status gets ignored on create so we need to update it else the InstallPlan sync ignores
+			// InstallPlans without any steps or bundle lookups
+			outIP.Status = ip.Status
+			Expect(ctx.Ctx().Client().Status().Update(context.Background(), outIP)).To(Succeed())
+
+			// The InstallPlan should be Failed due to the bundle unpack job timeout
+			// We wait for some time over the bundle unpack Job's timeout (i.e ActiveDeadlineSeconds) so that the Job can fail
+			waitFor := catalog.DefaultBundleUnpackTimeout + 30*time.Second
```

Comment:
oh I see the constant is being used in test code. Still, I'm not sure it's worth keeping around versus just declaring it in the test package here. Making a constant in live code to only use it in test code sounds a little strange

Reply:
Yeah my rationale for making it a constant was to always have the test time only after the bundle unpack timeout.

Else if we ever change the default for `--bundle-unpack-timeout` then you could have the test start failing as well unless you tweak the magic number here as well.

Reply:
Removed the constant for now.